keps/sig-api-machinery/4008-crd-ratcheting/README.md

```diff
@@ -1284,6 +1421,85 @@ The field to which the CEL rule in its `x-kubernetes-validations` list must
 contain all fields referred by the CEL rule; so the CEL error logic can have to 
 be modified to also include the rule's original location.
 
+### Different Ratcheting Rule Per Value Validation
+
+An alternative to the weaker form ratcheting described in the proposal would be
+to apply the DeepEqual rule to a smaller projection of the object being validated.
+
+i.e. MaxLength would use projection `len(obj)` since it only looks at that value,
+and if `len(oldObj) == len(newObj)`, then we would allow ratcheting.
+
+The following projections were considered:
+
+| Field               | Type     | Projection                   
+|---------------------|----------|------------------------------
+| Format              | string   | Identity                     
+| Maximum             | *float64  | Identity                    
+| ExclusiveMaximum    | bool     | Identity                    
+| Minimum             | *float64  | Identity                    
+| ExclusiveMinimum    | bool     | Identity                    
+| MaxLength           | *int64   | len(obj) -> int             
+| MinLength           | *int64   | len(obj) -> int             
+| Pattern             | string   | Identity                  
+| MultipleOf          | *float64  | Identity                 
+| Enum                | []JSON   | Identity                  
+| Nullable            | bool     | Identity                  
+| MaxItems            | *int64   | len(obj) -> int             
+| MinItems            | *int64   | len(obj) -> int             
+| UniqueItems         | bool     | Identity       
+| MaxProperties       | *int64   | len(obj) -> int             
+| MinProperties       | *int64   | len(obj) -> int             
+| Required            | []string | Keys(obj) -> []string (also can choose to ratchet on per-key presence)       
+
+
+#### Drawbacks
+
+This idea has a surprising UX. It makes no sense for `MaxLength: 3` to ratchet
+an update `hello` -> `fives`  but not `hello` -> `longer string`.
```

Comment:
Why not? What if MaxLength used to be 5?

Reply:
After offline discussion, we found that for our purposes there is no meaningful way to measure "closer" vs "farther" in the presence of CEL expressions, and without knowing the old schema.